internal/store/device.go

```diff
@@ -440,6 +441,17 @@ func (s *DeviceStore) Delete(ctx context.Context, orgId uuid.UUID, name string,
 	return nil
 }
 
+func (s *DeviceStore) SoftDelete(ctx context.Context, orgId uuid.UUID, name string) error {
```

Comment:
You're missing some logic that exists in Delete above.  I suggest changing that method instead of creating a new one by removing "Unscoped()" from innerTx.Unscoped().Delete(&existingRecord).

Reply:
+1. The current logic _should_ (let's verify!) already be a [soft-delete](https://gorm.io/docs/delete.html#Soft-Delete), no need to add something new here.

Reply:
@fzdarsky None of our DB delete methods use soft delete today.  Do we want to change them all, or only devices (and ERs)?

Reply:
I'm seeing that `store/device.go` the device uses [`Unscoped().Delete()`](https://github.com/flightctl/flightctl/blob/main/internal/store/device.go#L421) which is a hard delete. However I'm fine to change that method instead of creating this separate one.